### DISCUSSION TITLE
Only update package A when package B gets updated (even if package B is disabled in renovate)

### DISCUSSION THREAD

**Discussion by sammcj** [renovatebot/renovate#18034] [Request Help] [answered]
I'm managing NPM package upgrades for a number of repos that use AWS-CDK often with many other libs for applications developed within those same repos.

I have a group that contains `aws-cdk`, `aws-cdk-lib` and our internally developed `aws-cdk-internal*` packages.

This group is set to not upgrade because in order to do so - we first need to upgrade our aws-cdk-internal packages.

There is commonly used npm package called `constructs` - it isn't specific to aws-cdk, however - if it gets upgraded ahead of / out of alignment with what aws-cdk has, it can cause a lot of problems.

For this reason, I grouped the constructs package along with the aws-cdk packages - however renovate still recommends updating constructs despite the group being disabled (see below screenshot).

I've been reading through the renovate docs but am a little lost as to what I should be doing, and if perhaps I should be doing it in renovate at all and not in the package.json for npm somehow?

```json
    {
      "description": "We should match aws-cdk and constructs versions to versions used in aws-cdk-internal* packages",
      "matchPackagePatterns": [
        "aws-cdk",
        "aws-cdk-lib",
        "constructs",
        "^@our-org/aws-cdk-internal*"
      ],
      "groupName": "aws-cdk",
      "enabled": false,
      "addLabels": ["aws-cdk"]
    },
```

![Screen Shot 2022-09-29 at 8 21 04 am](https://user-images.githubusercontent.com/862951/192898993-1133c198-a6e4-46db-8c49-5f7f440940bc.jpg)

Apologies for the awkwardly worded question, I'm not quite sure of the term I'm looking for to link / set package dependencies if group isn't it, I hope it's not too confusing 😅

**Comment by rarkins** [comment, marked as answer]
I'm guessing you have more packageRules after the aws cdk one you configured, and one of them puts `constructs` constructs into the group in your screenshot. You either need to put your most important rules last, or make sure that rules which come after exclude package patterns you don't want to include.

**Reply by sammcj** [reply]
Thank you so much for the advice, moving the most important rules to last resolved the issue for me.

I almost wish there was a priority weighting you could provide to rules to make this more explicit.
